phpBB/includes/ucp/ucp_prefs.php

```diff
@@ -460,6 +461,132 @@ function main($id, $mode)
 					'S_SIG'		=> $data['sig'],
 					'S_NOTIFY'	=> $data['notify'])
 				);
+
+			break;
+
+			case 'wysiwyg':
+
+				$data = array(
+					'wysiwyg_editor'	=> $request->variable('wysiwyg_editor', $user->data['user_wysiwyg_editor']),
+				);
```

Comment:
I'm unsure about the code design in here. It doesn't seem to be correct. I've been thinking on many ways of doing this but none seem to be correct.
The objective is to show the options to the user according to the options he has chosen (if they are valid).
However, this code doesn't seem to be in the best way I could do this. Help and suggestions are appreciated.